src/main/java/com/amazon/jenkins/ec2fleet/EC2Api.java

```diff
@@ -0,0 +1,47 @@
+package com.amazon.jenkins.ec2fleet;
+
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.model.DescribeInstancesRequest;
+import com.amazonaws.services.ec2.model.DescribeInstancesResult;
+import com.amazonaws.services.ec2.model.Instance;
+import com.amazonaws.services.ec2.model.InstanceStateName;
+import com.amazonaws.services.ec2.model.Reservation;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class EC2Api {
+
+    public static Set<String> describeTerminated(final AmazonEC2 ec2, final Set<String> instanceIds) {
+        // assume all terminated until we get opposite info
+        final Set<String> terminated = new HashSet<>(instanceIds);
+        // don't do actual call if no data
+        if (instanceIds.isEmpty()) return terminated;
+
+        final DescribeInstancesRequest request = new DescribeInstancesRequest().withInstanceIds(instanceIds);
+
+        final Set<String> terminatedStates = new HashSet<>();
+        terminatedStates.add(InstanceStateName.Terminated.toString());
+        terminatedStates.add(InstanceStateName.Stopped.toString());
+        terminatedStates.add(InstanceStateName.Stopping.toString());
+        terminatedStates.add(InstanceStateName.ShuttingDown.toString());
+
+        DescribeInstancesResult result;
+        do {
+            result = ec2.describeInstances(request);
```

Comment:
Do we need to check case when it fail so we need to do that by instance instead of batch?